packages/react-scripts/scripts/build.js

```diff
@@ -122,9 +122,15 @@ checkBrowsers(paths.appPath, isInteractive)
       );
     },
     err => {
-      console.log(chalk.red('Failed to compile.\n'));
-      printBuildError(err);
-      process.exit(1);
+      const tscCompileOnError = process.env.TSC_COMPILE_ON_ERROR === 'true';
+      if (tscCompileOnError) {
+        console.log(chalk.yellow('Compiled with warnings.\n'));
```

Comment:
Maybe we should state here that due to the flag, these warnings may be serious. Or just say that there were errors that the user has chosen to ignore with the flag.

Reply:
I agree, we could make it more obvious that there were type errors on compilation. How about this:

```js
console.log(chalk.red(
  'Compiled with the following type errors (you may want to check these before deploying your app):\n'
));
```